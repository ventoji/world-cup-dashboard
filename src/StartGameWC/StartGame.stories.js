import React from 'react';
import StartGameWC from './StartGameWC';
import { action } from '@storybook/addon-actions';
import { withKnobs } from '@storybook/addon-knobs';

export default {
  title: 'WC Components/StartGaneWC',
  component: StartGameWC,
  decorators: [withKnobs]
};

const Template = (args) => <StartGameWC {...args} />;

export const Default = Template.bind({});


Default.args = {
  onSubmit: action('cick')
};