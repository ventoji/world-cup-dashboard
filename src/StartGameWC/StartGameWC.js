import React, { useState } from 'react';
import InputWC from '../InputWC';
import ButtonWC from '../ButtonWC';
import PropTypes from 'prop-types';

const INITIAL_GAME = {
  homeTeam: '',
  awayTeam: '',
  homeScore: 0,
  awayScore: 0,
}

const StartGameWC = ({ onSubmit }) => {
  const [matchDetails, setMatchDetails] = useState(INITIAL_GAME );

  const handleMatchDetails = (event) => {
    setMatchDetails({
      ...matchDetails,
      [event.target.name]: event.target.value,
    });
  };

  const sendMatchDetails = (event) => {
    event.preventDefault();
    onSubmit(matchDetails);
  };
  return (
    <>
      <InputWC
        htmlId="homeTeam"
        label="home team"
        type="text"
        placeholder="country name"
        name="homeTeam"
        onChange={handleMatchDetails}
      />
      <InputWC
        htmlId="awayTeam"
        label="away team"
        type="text"
        name="awayTeam"
        placeholder="country name"
        onChange={handleMatchDetails}
      />
      <ButtonWC label="Start new game" onClick={sendMatchDetails} />
    </>
  );
};

StartGameWC.propTypes = {
    /** Function to submit new players */
    onSubmit: PropTypes.func.isRequired
};

export default StartGameWC;
