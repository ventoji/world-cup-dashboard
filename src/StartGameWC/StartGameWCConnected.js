import React from 'react';
import { connect } from 'react-redux';
import StartGameWC from './StartGameWC';
import { addMatchWC } from '../store/actions/actions';
import { nameCountries } from '../utils';

export const StartGameWCConnected = ({ countries, addMatchWC }) => {
  const onSubmit = (mathDetails) => {
    addMatchWC(mathDetails);
  };
  return <StartGameWC onSubmit={onSubmit} countries={countries} />;
};

const mapStatetoProps = (state) => {
  return {
    countries: nameCountries(state.countries),
  };
};
const mapDispatchToProps = (dispatch) => ({
  addMatchWC: (mathDetails) => dispatch(addMatchWC(mathDetails)),
});

export default connect(
  mapStatetoProps,
  mapDispatchToProps,
)(StartGameWCConnected);
