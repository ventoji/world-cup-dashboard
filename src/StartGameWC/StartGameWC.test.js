import React from 'react';
import StartGameWC from './StartGameWC';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('StartGameWC component', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      onSubmit: jest.fn(),
      countries: ['Italy', 'Spain'],
    };

    component = shallow(<StartGameWC {...props} />);
  });

  it('Should be defined', () => {
    expect(component).toBeDefined();
  });

  it('should render properly', () => {
    expect(toJson(component)).toMatchSnapshot();
    expect(component.find('InputWC')).toHaveLength(2);
    expect(component.find('ButtonWC')).toHaveLength(1);
  });

  it('should submit a new game when button is clicked', () => {
    component = shallow(<StartGameWC {...props} />);
    //    expect(toJson(component)).toMatchSnapshot();
    component.find('#btn-start-game').simulate('click', {
      preventDefault: () => {},
    });

    //  console.log(component.find('ButtonWC'));
    expect(component.find('#btn-start-game')).toHaveLength(1);
    //     console.log(toJson(component).props.onSubmit.mock.calls.length);
    //expect(component.props.sendMatchDetails).toHaveBeenCalled();
 //   console.log(props.onSubmit.mock);
    //  expect(props.onSubmit.mock.calls.length).toEqual(1);
    //   expect(props.onSubmit).toHaveBeenCalledTimes(1);
  });
});
