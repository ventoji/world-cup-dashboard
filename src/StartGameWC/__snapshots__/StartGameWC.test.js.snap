// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartGameWC component should render properly 1`] = `
<Fragment>
  <InputWC
    htmlId="homeTeam"
    label="home team"
    name="homeTeam"
    onChange={[Function]}
    placeholder="country name"
    required={true}
    type="text"
  />
  <InputWC
    htmlId="awayTeam"
    label="away team"
    name="awayTeam"
    onChange={[Function]}
    placeholder="country name"
    required={true}
    type="text"
  />
  <ButtonWC
    id="btn-start-game"
    label="Start new game"
    onClick={[Function]}
  />
</Fragment>
`;
